src/chess/position.cc

```diff
@@ -30,14 +30,24 @@
 
 namespace lczero {
 
+// by default DrawMoveRule is 100 ply
+// this can be overwritten from Params initialization
+// for example when set to 40 the no_capture_ply will
+// be positioned so that the position believes there are only
+// 40 ply left before draw kicks in instead of 100
+//int DrawMoveRule = 100;
+// Number of moves till we call it a draw
+ int Position::drawmoverule_ = 100;
+
+
 Position::Position(const Position& parent, Move m)
     : no_capture_ply_(parent.no_capture_ply_ + 1),
       ply_count_(parent.ply_count_ + 1) {
   them_board_ = parent.us_board_;
   const bool capture = them_board_.ApplyMove(m);
   us_board_ = them_board_;
   us_board_.Mirror();
-  if (capture) no_capture_ply_ = 0;
+  if (capture) no_capture_ply_ = 0 + (100 - drawmoverule_);
```

Comment:
Why is this moved in such a weird way (setting draw to 20 no-capture-moves is done by setting no_capture_ply_ to 60 ply), instead of just changing it in `SearchWorker::ExtendNode`?

If you do that for NN to see high values when encoding the position, it's better to do that in position encoding function (as a separate or the same parameter) rather than changing Position class.

Reply:
I don't know anymore, this PR is from last year. I'll close it.